pkgs/development/tools/parsing/tree-sitter/update_impl.py

```diff
@@ -0,0 +1,133 @@
+from urllib.parse import quote
+import json
+import subprocess as sub
+import os
+import sys
+from typing import Generator, Any, Literal
+
+debug: bool = True if os.environ.get("DEBUG", False) else False
```

Comment:
I like the usage of argparse: it allows to self document the usage of the script.